Bottom: a047af98a6e61438b58178783ca67e7b9bc3311d
Top:    a65754bddea4f45b85f45d341fc7da431f05beca
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:58 +0000

polygon.c: Temporarily disable coalescing adjacent line segments

We need to extend the tests to properly account for arc segments.


---

diff --git a/src/polygon.c b/src/polygon.c
index fa4531b..9f73d09 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1364,6 +1364,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
   if (Undoing ())
     return (false);
 
+if (0) {
   for (n = 0; n < Polygon->PointN; n++)
     {
       prev = prev_contour_point (Polygon, n);
@@ -1373,6 +1374,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
       line.Point1 = Polygon->Points[prev];
       line.Point2 = Polygon->Points[next];
       line.Thickness = 0;
+#warning THIS TEST IS NOT SUFFICIENT.. DOESNT ACCOUNT FOR DIFFERENT ARC CENTERS
       if (Polygon->Points[prev].included_angle == Polygon->Points[n].included_angle &&
           IsPointOnLine (p->X, p->Y, 0.0, &line))
         {
@@ -1380,6 +1382,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
           changed = true;
         }
     }
+}
   return (changed);
 }
